clippy_lints/src/types/mod.rs

```diff
@@ -487,11 +525,21 @@ impl Types {
             _ => {},
         }
     }
+
+    /// This function checks if the type is allowed to change in the current context
+    /// based on the `avoid_breaking_exported_api` configuration
+    fn does_api_allow_change(&self, context: CheckTyContext) -> bool {
```

Comment:
Nit: `is_api_change_allowed`?

Reply:
`is_api_change_allowed` sounds to me like the function returns `true` if the API is allowed to change regardless of the current context. But the function is actually asking if the current context allows the change of the type. That's what I was trying to capture.

While typing this, I came up with the name `is_type_change_allowed()` which would describe what I wanted to say :upside_down_face:

Reply:
My confusion was that `does_api_allow_change` sounds like there is something about the API itself that doesn't allow changes. But in reality it's the clippy config that disallows changes. But I don't think the function name needs to communicate anything about how the return value is determined.